verifier/verifier.go

```diff
@@ -518,6 +536,72 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r, err := revocation.New(client)
+	if err != nil {
+		logger.Debug("error while creating a revocation client, err: %s", err.Error())
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  fmt.Errorf("unable to check revocation status, err: %s", err.Error()),
+		}
+	}
+
+	signingTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	revocationResult, err := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, signingTime)
```

Comment:
I think the only time this can happen is if for some reason the SingingTime isn't specified. I think this check makes sense, but I'm not 100% certain on the desired alternative behavior. For now, I'll update signingTime to time.Now() if IsZero() is true. That way the cert that has been revoked would be considered revoked regardless of the time it is RevokedAt (unless RevokedAt has been set to a time in the future, in which case this would not be considered revoked since it is before).

Reply:
> I think the only time this can happen is if for some reason the SingingTime isn't specified. I think this check makes sense, but I'm not 100% certain on the desired alternative behavior. For now, I'll update signingTime to time.Now() if IsZero() is true. That way the cert that has been revoked would be considered revoked regardless of the time it is RevokedAt (unless RevokedAt has been set to a time in the future, in which case this would not be considered revoked since it is before).

How about checking `IsZero()` in the notation-core-go code? something like:
```
...
if signingTime.IsZero() {xxx}
// Check if SigningTime was before the revocation
if opts.SigningTime.Before(resp.RevokedAt) {
	return errToServerResult(server, nil)
}
...
```

Reply:
I think that may obscure the behavior in notation-core-go. IMO we should keep notation-core-go the same (to avoid assuming how this zero time should be handled) as users can simply pass a different time if they want it handled differently (similar to the reassignment of signingTime that is being done in the current implementation).